### ISSUE TITLE
Autocomplete Errors

### ISSUE THREAD

**Issue by burrm** [rstudio/rstudio#13208]
### System details

    RStudio Edition : Desktop
    RStudio Version : RStudio 2023.03.1+446 "Cherry Blossom" Release (6e31ffc3ef2a1f81d377eeccab71ddc11cfbd29e, 2023-05-09) for windows Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.03.1+446 Chrome/108.0.5359.179 Electron/22.0.3 Safari/537.36
    OS Version :  Windows 10
    R Version :  R version 4.3.0 (2023-04-21 ucrt) -- "Already Tomorrow"

### Steps to reproduce the problem

1. Install R
2. Install RStudio
3. Tools -> Install Packages, enter 'roxygen2 devtools'
4. Load devtools package
5. install_github("burrm/lolcat")
6. Load lolcat package
7. Type 'lolcat::' when editing a script to see error (also reproducible with editing R script sections of RMarkdown)

### Describe the behavior you expected and the problem in detail

Should show autocomplete for package functions, instead fails with 

```
Error in order(typeScores, scores) : argument lengths differ
In addition: Warning message:
In completions$type == .rs.acCompletionTypes$DATAFRAME & !completions$context %in%  :
  longer object length is not a multiple of shorter object length
```
Interestingly, function parameters still autocomplete as expected, so this is limited to listing package functions with '::' operator when editing an R script.

Looking at other issues, others are looking at the issue with auto complete for large data frames instead of packages themselves, ex:

https://community.rstudio.com/t/error-in-newest-r-studio-for-dataframes-larger-than-2000-columns/165597
https://github.com/rstudio/rstudio/issues/12908
https://github.com/rstudio/rstudio/issues/13018 

Thanks for your time and effort looking at this. Please let me know if you need more information.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by burrm** [comment]
Looking at newest daily build, RStudio 2023.06.0+422 "Mountain Hydrangea" Release (25658d2cff5d0240fc09d6b7ddc2bc6e351ccfdf, 2023-06-06) for windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.06.0+422 Chrome/110.0.5481.208 Electron/23.3.0 Safari/537.36, looks like it is working.

When do you think that 2023.06.0+422 or a version based on this daily build will be released to end users?

**Comment by sharon-wang** [comment]
@burrm `2023.06.0+421` has been released and is available for download here: https://posit.co/download/rstudio-desktop/#download. The only difference between the released `+421` build and the `+422` build you tried is a minor release notes documentation update, so you should be good to go with the new release!

**Comment by sharon-wang** [comment]
Closing as this issue was fixed in 2023.06!
